clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -522,7 +523,24 @@ private void doSend(ClientRequest clientRequest, boolean isInternalRequest, long
         long updatedNow = this.time.milliseconds();
         List<ClientResponse> responses = new ArrayList<>();
         handleCompletedSends(responses, updatedNow);
-        handleCompletedReceives(responses, updatedNow);
+
+        try {
+            handleCompletedReceives(responses, updatedNow);
+        } catch (StaleMetadataException e) {
+            // close the connections to the traitor brokers upon this StateMetadataException
+            List<String> currentNodes = new ArrayList<>();
```

Comment:
we want to log how many broker connections we closed, and how many valid brokers we are left with?